Tavisca.Bootcamp.LanguageBasics.Exercise3/Program.cs

```diff
@@ -28,20 +30,153 @@ static void Main(string[] args)
+                    case 't':   onList = findMin(calorie,onList);
+                                if(onList.Count>1)
+                                    return Selection(protein,carbs,fat,calorie,diet.Substring(1),onList);
+                                else
+                                    return onList[0];
+                    default:    return onList[0];
+                }
+            }else{
+                return onList[0];
+            }
+
         }
 
-        public static int[] SelectMeals(int[] protein, int[] carbs, int[] fat, string[] dietPlans)
-        {
+        public static List<int> findMax(int[] element, List<int> onList){
+            int maximum;
+            List<int> onListnew = new List<int>();
+            if(onList.Count>0){
+                maximum=element[onList[0]];
+                foreach (int index in onList)
+                {
+                    if(maximum<element[index])
+                        maximum=element[index];
+                }
+                foreach (int index in onList){
+                    if(maximum==element[index])
+                        onListnew.Add(index);
+                }
+            }else{
+                maximum = element[0];
+                for (int i = 1; i < element.Length; i++)
+                {
+                    if(maximum<element[i])
+                        maximum=element[i];
+                }
+                for (int i = 0; i < element.Length; i++)
+                {
+                    if(maximum==element[i])
+                        onListnew.Add(i);
+                }
+            }
+            return onListnew;
+        }
+
+        public static List<int> findMin(int[] element, List<int> onList){
+            int minimum;
+            List<int> onListnew = new List<int>();
+            if(onList.Count>0){
+                minimum=element[onList[0]];
+                foreach (int index in onList)
+                {
+                    if(minimum>element[index])
+                        minimum=element[index];
+                }
+                foreach (int index in onList){
+                    if(minimum==element[index])
+                        onListnew.Add(index);
+                }
+            }else{
+                minimum = element[0];
+                for (int i = 1; i < element.Length; i++)
+                {
+                    if(minimum>element[i])
+                        minimum=element[i];
+                }
+                for (int i = 0; i < element.Length; i++)
+                {
+                    if(minimum==element[i])
+                        onListnew.Add(i);
+                }
+            }
+            return onListnew;
+        }
+
+        public static int[] SelectMeals(int[] protein, int[] carbs, int[] fat, string[] dietPlans)
+        {
             // Add your code here.
-            throw new NotImplementedException();
+            int len = protein.Length;

```

Comment:
You can directly use `protein.Length` instead of declaring a local variable. It will make code more readable. Otherwise one has to go up and down to see what `len` means.